spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java

```diff
@@ -533,7 +533,8 @@ private boolean isReadable(BeanDescription beanDesc, BeanPropertyWriter writer)
 			// that's what the metadata generator does. This filter is not used if there
 			// is JSON metadata for the property, so it's mainly for user-defined beans.
 			return (setter != null) || ClassUtils.getPackageName(parentType).equals(ClassUtils.getPackageName(type))
-					|| Map.class.isAssignableFrom(type) || Collection.class.isAssignableFrom(type);
+					|| Map.class.isAssignableFrom(type) || Collection.class.isAssignableFrom(type)
+					|| parentType.isRecord();
```

Comment:
I'm not sure that this is the correct fix, unfortunately. From a nesting perspective, records and classes are the same and the existing comparison of the package names should be sufficient. I suspect that expanding it to whenever a parent is a record may cause problems for richer, non-scalar types that shouldn't be included but now will be.

Reply:
Thanks for looking so quickly into this, I'm not sure what to tell you as tbh not sure what this impl is trying to do in the first place - this fix was just aimed to make those fields visible in serialized json. Can you share some light on purpose of this class? What it is trying to filter?

Reply:
I went through binding logic and understood that records are handled with matching constructor parameters - and as I was using multiple records constructors I was missing `@ConstructorBinding` annotation on one of them. It's working fine now - closing PR.